polynote-kernel/src/main/scala/polynote/kernel/remote/DeploySparkSubmit.scala

```diff
@@ -34,15 +34,23 @@ object DeploySparkSubmit extends DeployCommand {
     serverArgs: List[String] = Nil
   ): Seq[String] = {
 
+    val versionConfig = nbConfig.scalaVersion.flatMap(scalaVersion => {
+      nbConfig.sparkTemplate
+        .map(_.versionConfigs)
+        .flatMap(versionConfig => versionConfig.flatMap(_.find(_.versionNumber == scalaVersion)))
+    })
+
     val sparkConfig = config.spark.map(_.properties).getOrElse(Map.empty) ++
+      versionConfig.map(_.versionProperties).getOrElse(Map.empty) ++
       nbConfig.sparkTemplate.map(_.properties).getOrElse(Map.empty) ++
       nbConfig.sparkConfig.getOrElse(Map.empty)
 
-    val sparkArgs = (sparkConfig - "sparkSubmitArgs" - "spark.driver.extraJavaOptions" - "spark.submit.deployMode" - "spark.driver.memory")
+    val sparkArgs = (sparkConfig - "sparkSubmitArgs"  - "spark_submit_args" - "spark.driver.extraJavaOptions" - "spark.submit.deployMode" - "spark.driver.memory")
```

Comment:
Do we still need to worry about `spark_submit_args` if we do https://github.com/polynote/polynote/pull/1432/files#r1491846064 ?